src/VirtualClient/VirtualClient.Dependencies/Packaging/SnapPackageInstallation.cs

```diff
@@ -0,0 +1,214 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace VirtualClient.Dependencies
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.DependencyInjection;
+    using Polly;
+    using VirtualClient.Common;
+    using VirtualClient.Common.Extensions;
+    using VirtualClient.Common.Telemetry;
+    using VirtualClient.Contracts;
+
+    /// <summary>
+    /// Provides functionality for downloading and installing Yum packages
+    /// on the system.
+    /// </summary>
+    public class SnapPackageInstallation : VirtualClientComponent
```

Comment:
Great job, now please add a unit test for this class to verify the command with different kinds of input, like one package, multiple packages, null, or duplicated packages. Also you could verify the command difference in different distro.